tests/modules/users/test_permissions.py

```diff
@@ -6,34 +6,106 @@
 from werkzeug.exceptions import HTTPException
 
 from app.modules.users import permissions
+from app.modules.users.permissions.types import AccessOperation
 
 
-def test_DenyAbortMixin():
+# helpers to make object and module reading and writing testing more intuitive
+class NotARealClass(object):
+    # fake class for the module tests as the code checks against None
+    pass
+
+
+def validate_can_read_object(obj):
+    with permissions.ObjectAccessPermission(obj=obj, action=AccessOperation.READ):
+        pass
+
+
+def validate_cannot_read_object(obj):
     with pytest.raises(HTTPException):
-        permissions.rules.DenyAbortMixin().deny()
+        with permissions.ObjectAccessPermission(obj=obj, action=AccessOperation.READ):
+            pass
+
+
+def validate_can_write_object(obj):
+    with permissions.ObjectAccessPermission(obj=obj, action=AccessOperation.WRITE):
+        pass
+
+
+def validate_cannot_write_object(obj):
+    with pytest.raises(HTTPException):
+        with permissions.ObjectAccessPermission(obj=obj, action=AccessOperation.WRITE):
+            pass
+
+
+def validate_can_delete_object(obj):
+    with permissions.ObjectAccessPermission(obj=obj, action=AccessOperation.DELETE):
+        pass
+
+
+def validate_cannot_delete_object(obj):
+    with pytest.raises(HTTPException):
+        with permissions.ObjectAccessPermission(obj=obj, action=AccessOperation.DELETE):
+            pass
+
+
+def validate_can_read_module(module):
+    with permissions.ModuleAccessPermission(module=module, action=AccessOperation.READ):
+        pass
+
+
+def validate_cannot_read_module(module):
+    with pytest.raises(HTTPException):
+        with permissions.ModuleAccessPermission(
+            module=module, action=AccessOperation.READ
+        ):
+            pass
+
+
+def validate_can_write_module(module):
+    with permissions.ModuleAccessPermission(module=module, action=AccessOperation.WRITE):
+        pass
 
 
-def test_WriteAccessRule_authenticated_user(authenticated_user_instance):
-    authenticated_user_instance.is_active = True
-    assert permissions.rules.WriteAccessRule().check() is True
-    authenticated_user_instance.is_active = False
-    assert permissions.rules.WriteAccessRule().check() is False
+def validate_cannot_write_module(module):
+    with pytest.raises(HTTPException):
+        with permissions.ModuleAccessPermission(
+            module=module, action=AccessOperation.WRITE
+        ):
+            pass
+
+
+def validate_can_delete_module(module):
+    with permissions.ModuleAccessPermission(module=module, action=AccessOperation.DELETE):
+        pass
+
+
+def validate_cannot_delete_module(module):
+    with pytest.raises(HTTPException):
+        with permissions.ModuleAccessPermission(
+            module=module, action=AccessOperation.DELETE
+        ):
+            pass
 
 
-def test_ActiveUserRoleRule_anonymous(anonymous_user_instance):
+def test_DenyAbortMixin():
```

Comment:
the diff has got confused. test_DenyAbortMixin and all the test_ActiveUser tests are unchanged